.changeset/fuzzy-eggs-kneel.md

```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+Fixes an issue in the rewriting logic, where old data weren't puged during the rewrite flow, causing some false positiving during the rendering phase.
```

Comment:
```suggestion
Fixes an issue in the rewriting logic where old data was not purged during the rewrite flow. This caused some false positives when checking the validity of URL path names during the rendering phase.
``` 
We probably don't even need the second sentence (is it correct? I tried to infer from the PR description.) But if we do keep it, I think it's helpful to explain *what* the false positives were, so that people can compare it to what was happening in their project if they experienced this error!

(Would they know that whatever problem they were experienced was because data wasn't being purged? Is that meaningful to them as they use it? Or, did they have some problem and that was the underlying cause? If so, then connecting it to the problem they experienced is helpful!)